examples/models/object_detection/faster_rcnn/train.py

```diff
@@ -25,25 +25,145 @@
+
+        # Computes 2D image_scale.
+        image_scale = scaled_size / image_size
+
+        # Selects non-zero random offset (x, y) if scaled image is larger than
+        # desired_size.
+        if random_jittering:
+            max_offset = scaled_size - desired_size
+            max_offset = tf.where(
+                tf.less(max_offset, 0), tf.zeros_like(max_offset), max_offset
+            )
+            offset = max_offset * tf.random.uniform(
+                [
+                    2,
+                ],
+                0,
+                1,
+                seed=seed,
+            )
+            offset = tf.cast(offset, tf.int32)
+        else:
+            offset = tf.zeros((2,), tf.int32)
+
+        scaled_image = tf.image.resize(
+            image, tf.cast(scaled_size, tf.int32), method=method
+        )
+
+        if random_jittering:
+            scaled_image = scaled_image[
+                offset[0] : offset[0] + desired_size[0],
+                offset[1] : offset[1] + desired_size[1],
+                :,
+            ]
+
+        output_image = tf.image.pad_to_bounding_box(
+            scaled_image, 0, 0, padded_size[0], padded_size[1]
+        )
+
+        image_info = tf.stack(
+            [
+                image_size,
+                tf.constant(desired_size, dtype=tf.float32),
+                image_scale,
+                tf.cast(offset, tf.float32),
+            ]
+        )
+        return output_image, image_info
+
+
+def resize_and_crop_boxes(boxes, image_scale, output_size, offset):
+    with tf.name_scope("resize_and_crop_boxes"):
+        # Adjusts box coordinates based on image_scale and offset.
+        boxes *= tf.tile(tf.expand_dims(image_scale, axis=0), [1, 2])
+        boxes -= tf.tile(tf.expand_dims(offset, axis=0), [1, 2])
+        # Clips the boxes.
+        boxes = clip_boxes(boxes, output_size)
+        return boxes
+
+
+def clip_boxes(boxes, image_shape):
+    if boxes.shape[-1] != 4:
+        raise ValueError(
+            "boxes.shape[-1] is {:d}, but must be 4.".format(boxes.shape[-1])
+        )
+
+    with tf.name_scope("clip_boxes"):
+        if isinstance(image_shape, list) or isinstance(image_shape, tuple):
+            height, width = image_shape
+            max_length = [height, width, height, width]
+        else:
+            image_shape = tf.cast(image_shape, dtype=boxes.dtype)
+            height, width = tf.unstack(image_shape, axis=-1)
+            max_length = tf.stack([height, width, height, width], axis=-1)
+
+        clipped_boxes = tf.math.maximum(tf.math.minimum(boxes, max_length), 0.0)
+        return clipped_boxes
+
+
+def get_non_empty_box_indices(boxes):
+    # Selects indices if box height or width is 0.
+    height = boxes[:, 2] - boxes[:, 0]
+    width = boxes[:, 3] - boxes[:, 1]
+    indices = tf.where(tf.logical_and(tf.greater(height, 0), tf.greater(width, 0)))
+    return indices[:, 0]
+
+
+def resize_fn(image, gt_boxes, gt_classes):
```

Comment:
Can this functionality be achieved with RandomCropAndResize? If not, should we either update that layer or create a new KPL which provides this functionality?

Reply:
After training many classification and detection models, I have came to conclusion that a RandomResizeAndCrop-like behavior is required, but it cannot be achieved with the layers we have today.
What I would really like to see is, once I get to train some segmentation models and achieve SOTA, we can try to discuss what the best RandomResizeAndCrop layer should look like.
Note: Also the layer we have today does not support native box format, the same issue with the above comment (as mentioned in the PR description)

Reply:
I think that we could re-evaluate if some layers need to have an attached policy:
- https://github.com/keras-team/keras-cv/issues/108#issuecomment-1040626481
- https://github.com/keras-team/keras-cv/issues/108#issuecomment-1043598240

Reply:
The resize is somewhat different than other augmentation layers we have, especially for auto/rand-aug, colorized ops are used extensively, and only 1-2 geometric ops are popular choices

Reply:
sgtm. Maybe let's put a TODO in the code here for now to migrate this to a KPL?

Reply:
Done.

Reply:
In the case we will want to re-evaluate some policy in the future, at least for the classification task there was a nice survey out recently:
https://arxiv.org/abs/2206.06544

Reply:
I don't have the permission level to assign it to you but it is here:
https://github.com/keras-team/keras-cv/issues/900